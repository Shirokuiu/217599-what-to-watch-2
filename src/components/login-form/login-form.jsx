import React from "react";
import PropTypes from "prop-types";

export const LoginForm = (props) => {
  const {onUserInput, onSubmit, validation} = props;

  return <form action="#" className="sign-in__form">
    {validation.message ? <div className="sign-in__message">
      <p>{validation.message}</p>
    </div> : null}
    <div className="sign-in__fields">
      <div className={validation.email ? `sign-in__field` : `sign-in__field sign-in__field--error`}>
        <input
          className="sign-in__input"
          type="email"
          placeholder="Email address"
          name="email"
          id="user-email"
          onChange={onUserInput}
        />
        <label className="sign-in__label visually-hidden" htmlFor="user-email">Email address</label>
      </div>
      <div className={validation.password ? `sign-in__field` : `sign-in__field sign-in__field--error`}>
        <input
          className="sign-in__input"
          type="password"
          placeholder="Password"
          name="password"
          id="password"
          onChange={onUserInput}
        />
        <label className="sign-in__label visually-hidden" htmlFor="user-password">Password</label>
      </div>
    </div>
    <div className="sign-in__submit">
      <button className="sign-in__btn" type="button" onClick={onSubmit}>Sign in</button>
    </div>
  </form>;
};

LoginForm.propTypes = {
  onUserInput: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  validation: PropTypes.shape({
    email: PropTypes.bool.isRequired,
    password: PropTypes.bool.isRequired,
    message: PropTypes.string
  })
};
