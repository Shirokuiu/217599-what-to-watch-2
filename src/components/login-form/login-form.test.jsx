import React from "react";
import renderer from "react-test-renderer";

import {LoginForm} from "./login-form";

it(`Is login form rendered`, () => {
  const tree = renderer.create(<LoginForm
    onUserInput={jest.fn()}
    onSubmit={jest.fn()}
    validation={{
      email: true,
      password: true,
      message: undefined
    }}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
