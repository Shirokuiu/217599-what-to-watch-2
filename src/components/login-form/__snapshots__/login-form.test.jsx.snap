// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Is login form rendered 1`] = `
<form
  action="#"
  className="sign-in__form"
>
  <div
    className="sign-in__fields"
  >
    <div
      className="sign-in__field"
    >
      <input
        className="sign-in__input"
        id="user-email"
        name="email"
        onChange={[MockFunction]}
        placeholder="Email address"
        type="email"
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="user-email"
      >
        Email address
      </label>
    </div>
    <div
      className="sign-in__field"
    >
      <input
        className="sign-in__input"
        id="password"
        name="password"
        onChange={[MockFunction]}
        placeholder="Password"
        type="password"
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="user-password"
      >
        Password
      </label>
    </div>
  </div>
  <div
    className="sign-in__submit"
  >
    <button
      className="sign-in__btn"
      onClick={[MockFunction]}
      type="button"
    >
      Sign in
    </button>
  </div>
</form>
`;
